stratum/hal/lib/common/p4_service.cc

```diff
@@ -833,42 +850,48 @@ bool P4Service::IsMasterController(uint64 node_id, uint64 connection_id) const {
   return it->second.begin()->connection_id() == connection_id;
 }
 
-void* P4Service::PacketReceiveThreadFunc(void* arg) {
-  auto* args = reinterpret_cast<ReaderArgs<::p4::v1::PacketIn>*>(arg);
+void* P4Service::StreamResponseReceiveThreadFunc(void* arg) {
+  auto* args =
+      reinterpret_cast<ReaderArgs<::p4::v1::StreamMessageResponse>*>(arg);
   auto* p4_service = args->p4_service;
   auto node_id = args->node_id;
   auto reader = std::move(args->reader);
   delete args;
-  return p4_service->ReceivePackets(node_id, std::move(reader));
+  return p4_service->ReceiveStreamRespones(node_id, std::move(reader));
 }
 
-void* P4Service::ReceivePackets(
-    uint64 node_id, std::unique_ptr<ChannelReader<::p4::v1::PacketIn>> reader) {
+void* P4Service::ReceiveStreamRespones(
+    uint64 node_id,
+    std::unique_ptr<ChannelReader<::p4::v1::StreamMessageResponse>> reader) {
   do {
-    ::p4::v1::PacketIn packet_in;
-    // Block on next packet RX from Channel.
-    int code = reader->Read(&packet_in, absl::InfiniteDuration()).error_code();
+    ::p4::v1::StreamMessageResponse resp;
+    // Block on next stream response RX from Channel.
+    int code = reader->Read(&resp, absl::InfiniteDuration()).error_code();
     // Exit if the Channel is closed.
     if (code == ERR_CANCELLED) break;
     // Read should never timeout.
     if (code == ERR_ENTRY_NOT_FOUND) {
       LOG(ERROR) << "Read with infinite timeout failed with ENTRY_NOT_FOUND.";
       continue;
     }
-    // Handle PacketIn.
-    PacketReceiveHandler(node_id, packet_in);
+    // Handle StreamMessageResponse.
+    StreamResponseReceiveHandler(node_id, resp);
   } while (true);
   return nullptr;
 }
 
-void P4Service::PacketReceiveHandler(uint64 node_id,
-                                     const ::p4::v1::PacketIn& packet) {
-  // We send the packets only to the master controller stream for this node.
+void P4Service::StreamResponseReceiveHandler(
+    uint64 node_id, const ::p4::v1::StreamMessageResponse& resp) {
+  // We don't expect arbitration updates from the switch.
+  if (resp.has_arbitration()) {
+    LOG(ERROR) << "Received MasterArbitrationUpdate from switch. This should "
```

Comment:
`FATAL`?

Reply:
Fatal is probably warranted here, but it's also not harmful to just drop the arb update and continue...

Reply:
We agreed on `FATAL` on the basis that this is a serious bug in our code and we want to increase the chances of getting a user report in that case.